### PR TITLE
feat: added cross account set up for aws registry

### PR DESCRIPTION
**Pull request by milan-deepfence**
Changes proposed in this pull request:
- Added cross account set up for AWS registry

@deepfence/engineering

Screenshot 2023-01-04 at 10 44 49 AM

### CODE REVIEW

File: deepfence_ui/app/scripts/actions/app-actions.js [hunk 1]
```diff
@@ -1371,6 +1372,15 @@ export function listRegistryImagesAction(params) {
   return genericThunkAction(actionTypes, listRegistryImages, params);
 }
 
+export function getAccountIdAction() {
```
Comment (manV) [inline deepfence_ui/app/scripts/actions/app-actions.js:1375]: maybe rename to `getConsoleAWSAccountId`

File: deepfence_ui/app/scripts/reducers/cve-reducer.js [hunk 2]
```diff
@@ -439,7 +465,8 @@ function containerImageRegistryReducer(state = Map({}), action) {
       const {
         payload: {
           success,
-          data = []
+          data = [],
+          account_id = null
```
Comment (manV) [inline deepfence_ui/app/scripts/reducers/cve-reducer.js:469]: is this needed now?

File: deepfence_ui/app/scripts/reducers/cve-reducer.js [hunk 3]
```diff
@@ -426,6 +426,32 @@ function containerImageRegistryReducer(state = Map({}), action) {
       return state;
     }
 
+    case ActionTypes.GET_ACCOUNT_ID_REQUEST: {
+      state = state.setIn(['ecr', 'data'], {});
```
Comment (manV) [inline deepfence_ui/app/scripts/reducers/cve-reducer.js:430]: also maybe change this path to `['meta', 'aws', 'accountId']` or something?

File: deepfence_ui/app/scripts/components/vulnerability-view/registry-scan/credentials-form.js [hunk 4]
```diff
@@ -132,16 +139,23 @@ class ContainerImageRegistryCredentialsForm extends React.PureComponent {
...
-                component={fieldTypeRenderer(field.type)}
-                fileName={fileValues[field.name] ? fileValues[field.name].name : ''}
-                labelBefore={field.label}
-                filenamePosition="before"
-                validate={validateJsonFileType}
-                {...field}
-              />
+              <>
+                {
+                  field.name === 'non_secret.target_account_role_arn' && enableField ?
```
Comment (manV) [inline deepfence_ui/app/scripts/components/vulnerability-view/registry-scan/credentials-form.js:144]: do we also need to check if the form is for ecr?

File: deepfence_ui/app/scripts/reducers/cve-reducer.js (hunk 2 above)
Comment (milan-deepfence) [reply deepfence_ui/app/scripts/reducers/cve-reducer.js]: removed

File: deepfence_ui/app/scripts/actions/app-actions.js (hunk 1 above)
Comment (milan-deepfence) [reply deepfence_ui/app/scripts/actions/app-actions.js]: renamed

File: deepfence_ui/app/scripts/components/vulnerability-view/registry-scan/credentials-form.js (hunk 4 above)
Comment (milan-deepfence) [reply deepfence_ui/app/scripts/components/vulnerability-view/registry-scan/credentials-form.js]: added

**Comment by manV** [comment]
Stale, Closing.

**Pull request closed without merging**